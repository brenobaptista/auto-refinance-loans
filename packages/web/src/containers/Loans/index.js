import React, { useState, useEffect } from 'react'
import Card from '../../components/Card'
import styles from './index.module.css'

const Loans = () => {
  const [loans, setLoans] = useState([])

  useEffect(() => {
    const getLoans = async () => {
      try {
        const response = await fetch('http://localhost:3333/loans')

        const loans = await response.json()

        setLoans(loans)
      } catch (error) {
        console.error(error)
      }
    }

    getLoans()
  }, [])

  return (
    <div className={styles.loans}>
      <div className={styles.title}>Select the loan you want to refinance</div>
      <div className={styles.loansWrapper}>
        {loans.map(
          ({
            id,
            lender,
            annualPercentage,
            monthlyPayment,
            paidPeriod,
            totalPeriod,
            vehicle
          }) => (
            <Card
              key={id}
              loanId={id}
              photo={`http://localhost:3333/vehicles/${vehicle.photo}`}
              info={`${vehicle.year} ${vehicle.make} ${vehicle.model}`}
              mileage={vehicle.mileage}
              lender={lender}
              annualPercentage={annualPercentage}
              monthlyPayment={monthlyPayment}
              paidPeriod={paidPeriod}
              totalPeriod={totalPeriod}
            />
          )
        )}
      </div>
    </div>
  )
}

export default Loans
