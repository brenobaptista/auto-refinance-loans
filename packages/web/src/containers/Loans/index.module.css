@keyframes roll-down {
  0% {
    opacity: 0;
    transform: translate(0, -16px);
  }

  100% {
    opacity: 1;
    transform: translate(0);
  }
}

.loans {
  display: flex;
  justify-content: center;
  padding: 0 24px;
  animation: roll-down 1s;
}

.wrapper {
  max-width: 400px;
}

.title {
  margin: 16px 0 48px;
  font-size: 1.5rem;
  font-weight: 600;
}
