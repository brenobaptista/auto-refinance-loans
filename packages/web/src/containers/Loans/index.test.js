import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Loans from "./index";

beforeEach(() => {
  fetch.resetMocks();
});

test("matches snapshot", async () => {
  fetch.mockResponseOnce(
    JSON.stringify([
      {
        id: 1,
        lender: "The Big Bank",
        annualPercentage: 0.142,
        loanBalance: 15000,
        monthlyPayment: 377.5,
        paidPeriod: 18,
        totalPeriod: 72,
        vehicle: {
          photo: "1.jpg",
          make: "HONDA",
          model: "ACCORD",
          year: 2019,
          mileage: 41855,
        },
      },
    ])
  );

  const { asFragment, findByText } = render(<Loans />, {
    wrapper: MemoryRouter,
  });

  const vehicle = await findByText("2019 HONDA ACCORD");

  expect(vehicle).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});

test("matches snapshot when expection", async () => {
  fetch.mockReject(() => Promise.reject("API is down"));

  const { asFragment } = render(<Loans />, {
    wrapper: MemoryRouter,
  });

  expect(asFragment()).toMatchSnapshot();
});
