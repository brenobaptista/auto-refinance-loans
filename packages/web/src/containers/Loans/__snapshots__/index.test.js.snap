// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="loans"
  >
    <div>
      <div
        class="title"
      >
        Select the loan you want to refinance
      </div>
      <div
        class="loansWrapper"
      >
        <div
          class="card"
        >
          <img
            alt="Vehicle"
            class="vehicleImage"
            src="http://localhost:3333/vehicles/1.jpg"
          />
          <div
            class="cardInfo"
          >
            <div
              class="vehicleInfo"
            >
              <div
                class="textImportant"
              >
                2019 HONDA ACCORD
              </div>
              <div
                class="vehicleMileage"
              >
                with an estimated 
                <span
                  class="textImportant"
                >
                  41855
                </span>
                 miles
              </div>
            </div>
            <div
              class="loanInfo"
            >
              <div>
                Lender
              </div>
              <div
                class="textImportant"
              >
                The Big Bank
              </div>
            </div>
            <hr />
            <div
              class="loanInfo"
            >
              <div>
                APR
              </div>
              <div
                class="textImportant"
              >
                14.20%
              </div>
            </div>
            <hr />
            <div
              class="loanInfo"
            >
              <div>
                Monthly payment
              </div>
              <div
                class="textImportant"
              >
                $377.5
              </div>
            </div>
            <hr />
            <div
              class="loanInfo"
            >
              <div>
                Time remaining
              </div>
              <div
                class="textImportant"
              >
                54/72 mo
              </div>
            </div>
            <a
              href="/offers/1"
            >
              <button
                class="btn"
                type="button"
              >
                Select loan
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when expection 1`] = `
<DocumentFragment>
  <div
    class="loans"
  >
    <div>
      <div
        class="title"
      >
        Select the loan you want to refinance
      </div>
      <div
        class="loansWrapper"
      />
    </div>
  </div>
</DocumentFragment>
`;
